Scripts/RemediationScripts/Remediate-ConfigureIoTHubTLSVersion.ps1

```diff
@@ -0,0 +1,878 @@
+                    $IoTHubDetail = Fetch-API -Method Get -Uri $iotHubResourceAPI -ErrorAction Stop
+
+            
+                    $IoTHubDetails += $IoTHubDetail | Select-Object @{N='ResourceId';E={$_.Id}},
+                                                                     @{N='ResourceGroupName';E={$_.ResourceGroup}},
+                                                                     @{N='ResourceName';E={$_.Name}},
+                                                                     @{N='MinimumTlsVersion';E={$_.Properties.MinTlsVersion}},
+                                                                     @{N='Location';E={$_.Location}},
+                                                                     @{N='Sku';E={$_.Sku}}
+                }
+                catch
+                {
+                    Write-Host "Error fetching IoT Hub(s) resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                    $logResource = @{}
+                    $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                    $logResource.Add("ResourceName",($_.ResourceName))
+                    $logResource.Add("Reason","Error fetching IoT Hub(s) information.")    
+                    $logSkippedResources += $logResource
+                }
+            }
+        }
+    }
+
+    $totalIoTHub = ($IoTHubDetails | Measure-Object).Count
+
+    if ($totalIoTHub -eq 0)
+    {
+        Write-Host "No IoT Hub(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        break
+    }
+  
+    Write-Host "Found [$($totalIoTHub)] IoT Hub(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list for storing IoT Hub(s) where required TLS Version is not configured
+    $NonCompliantTLSIoTHub = @()
+
+    # Non Compliant IoT Hub(s) with TLS Version as default
+    $TLSwithDefaultValue = @()
+
+    Write-Host "Separating IoT Hub(s) for which TLS Version [$($requiredMinTLSVersion)] is not configured..."
+
+    $IoTHubDetails | ForEach-Object {
+        $IoTHub = $_
+        if(!$_.MinimumTlsVersion -or $_.MinimumTlsVersion -lt $requiredMinTLSVersion)
+        {
+            $NonCompliantTLSIoTHub += $IoTHub
+
+            if(!$_.MinimumTlsVersion){
+                $TLSwithDefaultValue += $IoTHub
+            }
+        }
+        else
+        {
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+            $logResource.Add("ResourceName",($_.ResourceName))
+            $logResource.Add("Reason","Minimum required TLS Version configured set on IoT Hub.")    
```

Comment:
Minimum required TLS Version configured set on IoT Hub-->Minimum required TLS Version is configured on IoT Hub

Reply:
Updated message as suggested